src/client/rooms/room-page.spec.js

```diff
@@ -5,30 +5,92 @@ import { mount, shallow } from 'enzyme';
 import RoomPage from './room-page';
 import store from '../store';
 
-var data;
 describe('<RoomPage />', () => {
-
+    let data;
+    let cloakService;
     beforeEach(() => {
         window.cloak = jasmine.createSpyObj('cloak', ['configure', 'run', 'connected', 'message']);
+        cloakService = jasmine.createSpyObj('cloakService',['configureAndRun', 'isConnected', 'messageJoinRoom', 'getRoomData',
+            'messageRemoveFromRoomList', 'messageLeaveRoom']);
+        RoomPage.__Rewire__('cloakService', cloakService);  
     });
 
-    it('contains "Start Game" button', () => {
+    afterEach(() => {
+        RoomPage.__ResetDependency__('cloakService');
+    })
+
+    it('contains "Start" button', () => {
         data = '';
-        const wrapper = mount(
+        let wrapper = mount(
             <Provider store={store}>
                 <RoomPage params={{data: data}}/>
             </Provider>
         );
-        expect(wrapper.find('#start-game').text()).toEqual('Start Game');
+
+        expect(wrapper.find('#start-game').text()).toEqual('Start');
     });
 
-    it('contains "Leave Room" button', () => {
+    it('contains "Leave" button', () => {
         data = '';
-        const wrapper = mount(
+        let wrapper = mount(
             <Provider store={store}>
                 <RoomPage params={{data: data}}/>
             </Provider>
         );
-        expect(wrapper.find('#leave-room').text()).toEqual('Leave Room');
+
+        expect(wrapper.find('#leave-room').text()).toEqual('Leave');
+    });
+
+    describe('componentWillMount', () => {
+        it('if user is connected to the server it tries to join the room', () => {
+            data = '1234';
+            cloakService.isConnected.and.returnValue(true);
+            let wrapper = mount(
+                <Provider store={store}>
+                    <RoomPage params={{data: data}}/>
+                </Provider>
+            );
+
+            expect(cloakService.messageJoinRoom).toHaveBeenCalledWith(data);
+        });
+
+        it('if user is connected to the server it tries to get the room data', () => {
+            data = '1234';
+            cloakService.isConnected.and.returnValue(true);
+            let wrapper = mount(
+                <Provider store={store}>
```

Comment:
For future reference - nothing to change here - if you export the main class as well as the default export, you can then use `import { CLASS } from 'FILE';`; then you don't need to use the Provider or store any more